core/meterer/onchain_state.go

```diff
@@ -28,17 +30,18 @@ var _ OnchainPayment = (*OnchainPaymentState)(nil)
 type OnchainPaymentState struct {
 	tx *eth.Reader
 
-	ActiveReservations map[string]core.ActiveReservation
-	OnDemandPayments   map[string]core.OnDemandPayment
+	ActiveReservations *map[gethcommon.Address]core.ActiveReservation
```

Comment:
you probably want `map[gethcommon.Address]*core.ActiveReservation`

Reply:
yes, thank you